### PR TITLE
Enhanced ReplayDataParser with ModeratorEvent Support

### LINKED ISSUE #121: Add support for the `ModeratorEvent` sim callback
## Description

The goal is to introduce a new sim callback with the sole responsibility to inform the moderators. At the moment we (ab)use the `GiveResourcesToPlayer` callback which is also (ab)used by chat messages. Therefore we'll effectively extend the functionality of the following file/class:

- faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java

## Checklist

- [x] Create a new class called `ModeratorEvent`, similar to [ChatMessage](https://github.com/FAForever/faf-java-commons/blob/develop/faf-commons-data/src/main/java/com/faforever/commons/replay/ChatMessage.java) except that there's no receiver field.
- [x] Create a new private field `ModeratorEvents` in the [ReplayDataParser](https://github.com/FAForever/faf-java-commons/blob/develop/faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java) class. This will be an array list of type `ModeratorEvent`, just like the chat messages are.
- [x] In the function `parseTicks` we want to extend the case `CMDST_LUA_SIM_CALLBACK` to also check for the `ModeratorEvent` sim callback. The format of the sim callback is as follows:

```lua
---@class ModeratorEvent
---@field From number
---@field Message string
```

And as an example:

```lua
{ From: 1, Message: 'killed 50 units' }
```

These should be parsed and pushed into the previously mentioned private field `ModeratorEvents`. An (untested) example can be found in: https://github.com/FAForever/faf-java-commons/pull/102

## Notes

It may be interesting to also store the active command source in the `ModeratorEvent` class. If the player in question is not meddling with the callbacks, it should represent the same value as the `From` field in the callback.

### PR DESCRIPTION
**Pull request by magge-faf**
Thanks, @Garanas for the detailed task issue and the examples. Helped me tremendously.

My steps:

1. Created a FAF Develop replay (TestModeratorEvents.fafreplay) with numerous actions (chatting, marker, ping commands, ctrlk+k unit)
2. New class `ModeratorEvent` 
3. New private field `ModeratorEvents` to store the `ModeratorEvent` objects
4. Check for the `ModeratorEvent` in `CMDST_LUA_SIM_CALLBACK`
5. Created a method `parseModeratorEvent`
6. Added unit test to verify that `parseModeratorEvent` correctly parses the replay. (Also, I just console printed out the results I got from parsing and checked it manually first just to be sure)

Output in console:

image

Example was :

`{ From: 1, Message: 'killed 50 units' }`

We have access to the `ModeratorEvents` which are stored in the replay file, and we can build on top of that now.

I have already added the `activeCommandSource` for the future features that will depend on that.

If something needs to be changed, please don't hesitate to comment.

### CODE REVIEW

File: faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java [hunk 1]
```diff
@@ -375,6 +377,18 @@ private void parseGiveResourcesToPlayer(Map<String, Object> lua) {
     }
   }
 
+
+  void parseModeratorEvent(Map<String, Object> lua) {
+    String messageContent = (String) lua.get("Message");
+    Float fromFloat = (Float) lua.get("From");
```
Comment (Garanas) [inline faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java]: These should be integers, as they represent the focus army.

File: faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java [hunk 2]
```diff
@@ -375,6 +377,18 @@ private void parseGiveResourcesToPlayer(Map<String, Object> lua) {
...
+    int activeCommandSource = 0; // activeCommandSource is not available in TestModeratorEvents.fafreplay, yet
+    if (lua.containsKey("activeCommandSource")) {
+      activeCommandSource = ((Number) lua.get("activeCommandSource")).intValue();
+    }
```
Comment (Garanas) [inline faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java]: This is not how it works, but it's also not blocking the pull request. We can look into this the next iteration.

**Review by Garanas — APPROVED** [review]
To me this implementation looks like a solid start and can be merged on its own. Perhaps @Sheikah45 and/or @Brutus5000 has additional insights and/or comments

File: faf-commons-data/src/main/java/com/faforever/commons/replay/ModeratorEvent.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import java.time.Duration;
+
+@Data
+public class ModeratorEvent {
+
+  private final Duration time;
+  private final String sender;
+  private final String message;
+  private final int activeCommandSource;
+}
```
Comment (Sheikah45) [inline faf-commons-data/src/main/java/com/faforever/commons/replay/ModeratorEvent.java]: This class can just be a record since all fields are final

**Review by Sheikah45 — CHANGES_REQUESTED** [review]

**Review by Sheikah45 — APPROVED** [review]

**Pull request merged**